tasks/setup.py

```diff
@@ -277,6 +266,353 @@ def setupPulumiConfig(config):
+    config_path: Optional[str] = None,
+) -> None:
+    """
+    Create a new key pair using the AWS CLI, save the key pair to disk, and update the local config.
+
+    This task is interactive and can be run with no options:
+      - Prompt user for missing values
+      - Confirm overwrite of existing key pair
+      - Confirm update of local config
+    """
+    try:
+        config = get_local_config(config_path)
+    except Exception as e:
+        error(f"{e}")
+        error("Failed to load config")
+        raise Exit(code=1)
+
+    _aws_create_keypair(
+        ctx=ctx,
+        config=config,
+        keypair_name=keypair_name,
+        key_type=key_type,
+        private_key_path=private_key_path,
+        public_key_path=public_key_path,
+        use_aws_vault=use_aws_vault,
+        aws_account_name=aws_account_name,
+    )
+
+    update_config_aws_keypair(
+        config,
+        config_path=config_path,
+    )
+
+
+def _aws_create_keypair(
+    ctx: Context,
+    config: Config,
+    keypair_name: Optional[str] = None,
+    key_type: Optional[str] = None,
+    private_key_path: Optional[str] = None,
+    public_key_path: Optional[str] = None,
+    use_aws_vault: Optional[bool] = False,
+    aws_account_name: Optional[str] = None,
+) -> None:
+    if config.configParams is None or config.configParams.aws is None:
+        raise Exit("Config is missing aws section")
+
+    keypair_opts = resolve_keypair_opts(
+        config=config,
+        keypair_name=keypair_name,
+        key_type=key_type,
+        private_key_path=private_key_path,
+        public_key_path=public_key_path,
+        require_key_type=True,
+        require_keyfile_exists=False,
+    )
+    keypair_name = str(keypair_opts["keypair_name"])
+    key_type = str(keypair_opts["key_type"])
+    private_key_path = str(keypair_opts["private_key_path"])
+    public_key_path = str(keypair_opts["public_key_path"])
+
+    def _get_aws_cmd(cmd):
+        return get_aws_cmd(cmd, use_aws_vault=use_aws_vault, aws_account=aws_account_name)
+
+    # check if key pair already exists
+    if not check_existing_aws_keypair(
+        ctx, keypair_name, use_aws_vault=use_aws_vault, aws_account_name=aws_account_name
+    ):
+        return
+    if Path(private_key_path).exists():
+        warn(f"Private key {private_key_path} already exists.")
+        if not ask_yesno("Do you want to overwrite it?"):
+            return
+        # delete existing key pair
+        os.remove(private_key_path)
+
+    # generate private key
+    cmd = f'ec2 create-key-pair --key-name "{keypair_name}" --key-type {key_type} --query KeyMaterial --output text'
+    out = ctx.run(_get_aws_cmd(cmd), hide=True)
+    if out is None:
+        raise Exit(f"Failed to create key pair {keypair_name}")
+    key_material = out.stdout.strip()
+    # write private key to disk
+    os.makedirs(Path(private_key_path).parent, exist_ok=True)
+    with open(private_key_path, "w") as f:
+        f.write(key_material)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
There are a couple of missing steps:
- adding a passkey to protect the private key, or ssh scanner will complain
- adding the key to ssh-agent, at list on unix environments

Reply:
> adding a passkey to protect the private key, or ssh scanner will complain

There are some weird synergies between us, pulumi, and AWS, here., i've previously documented them [here](https://github.com/DataDog/test-infra-definitions/blob/b7042cd9a532218fb9bb6f787cd020948b4b9786/tasks/setup.py#L506-L525).
- if you have an encrypted `privateKeyPath` configured, but don't provide `privateKeyPassword`, then pulumi will fail and won't check ssh-agent.
- our `host_ssh.go` code looks similar, but doesn't append the auth methods so will skip ssh-agent if `privateKeyPath` is configured.
- pulumi doesn't support windows ssh-agent so we need to `privateKeyPath` and if encrypted `privateKeyPassword`
- if you don't have `privateKeyPath` configured, or if it is encrypted, you can't RDP to Windows hosts / decrypt their password

I think they're all surmountable, but will require changes in other places. For example, our RDP helper task could prompt the user for the passphrase and temporarily decrypt the key when requesting the password from AWS.

Reply:
We could also create the key on the fly as @vboulineau suggested a long time ago, at VM creation, and then remove the ssh key both locally and remote. This way DD ssh key monitor doesn't bother us, as they are ephemeral.

Reply:
It would remove the need for this setup. We would keep only the key for the CI, stored in 1Password and in `AWS`

Reply:
that would be neat!